### PR TITLE
Refactored code

### PR DESCRIPTION
**Pull request by DGolubets**

### CODE REVIEW

File: src/slack.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use failure::Error;
+use model::*;
+use slack_hook::{PayloadBuilder, Slack, SlackLink};
+use slack_hook::SlackTextContent::{Link, Text};
+
+pub fn notify_release(slack: &Slack, channel: &str, emoji: &str, repo: &GithubRepo, tags: &[Tag]) -> Result<(), Error> {
+    for tag in tags {
+        notify_release_single(slack, channel, emoji, repo, &tag.name)?;
```
Comment (grillorafael) [inline src/slack.rs]: Can we merge this into one slack message? for all tags

File: src/main.rs [hunk 2]
```diff
@@ -1,149 +1,77 @@
...
-                },
-                Err(_error) =>
-                    println!("Error querying {}", repo),
-            }
+        for repo in &github_repos {
+            debug!("Getting tags for {}", repo);
+            get_github_tags(&github_client, &repo).and_then(|tags| {
+                debug!("Tags for {}: {:?}", repo, tags);
+                let new_tags = tag_store.update(&repo, tags);
+                notify_release(&slack, &slack_channel, &slack_emoji, &repo, &new_tags)
```
Comment (grillorafael) [inline src/main.rs]: On startup will it notify all tags? If so, let's notify only new ones (i.e. first pass won't notify anything)

Comment (DGolubets) [reply src/main.rs]: no, it notifies only second time

Comment (grillorafael) [reply src/main.rs]: :+1:

**Pull request merged**